src/cdk-experimental/menu/context-menu.spec.ts

```diff
@@ -0,0 +1,314 @@
+import {Component, ViewChild, ElementRef} from '@angular/core';
+import {CdkMenuModule} from './menu-module';
+import {TestBed, async, ComponentFixture} from '@angular/core/testing';
+import {CdkMenu} from './menu';
+import {CdkContextMenuTrigger} from './context-menu';
+import {dispatchMouseEvent} from '@angular/cdk/testing/private';
+import {By} from '@angular/platform-browser';
+import {CdkMenuItem} from './menu-item';
+import {CdkMenuItemTrigger} from './menu-item-trigger';
+
+describe('CdkContextMenuTrigger', () => {
+  describe('with simple context menu trigger', () => {
+    let fixture: ComponentFixture<SimpleContextMenu>;
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [CdkMenuModule],
+        declarations: [SimpleContextMenu],
+      }).compileComponents();
+    }));
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(SimpleContextMenu);
+      fixture.detectChanges();
+    });
+
+    /** Get the menu opened by the context menu trigger. */
+    function getContextMenu() {
+      return fixture.componentInstance.menu;
+    }
+
+    /** Return a reference to the context menu native element. */
+    function getNativeContextMenu() {
+      return fixture.componentInstance.nativeMenu?.nativeElement;
+    }
+
+    /** Get the context in which the context menu should trigger. */
+    function getMenuContext() {
+      return fixture.componentInstance.trigger.nativeElement;
+    }
+
+    /** Open up the context menu and run change detection. */
+    function openContextMenu() {
+      // right click triggers a context menu event
+      dispatchMouseEvent(getMenuContext(), 'contextmenu');
+      fixture.detectChanges();
+    }
+
+    it('should display context menu on right click inside of context component', () => {
+      expect(getContextMenu()).not.toBeDefined();
+      openContextMenu();
+      expect(getContextMenu()).toBeDefined();
+    });
+
+    it('should close out the context menu when clicking in the context', () => {
+      openContextMenu();
+
+      getMenuContext().click();
+      fixture.detectChanges();
+
+      expect(getContextMenu()).not.toBeDefined();
+    });
+
+    it('should close out the context menu when clicking on element outside of the context', () => {
+      openContextMenu();
+
+      fixture.nativeElement.querySelector('#other').click();
+      fixture.detectChanges();
+
+      expect(getContextMenu()).not.toBeDefined();
+    });
+
+    it('should close out the context menu when clicking a menu item', () => {
+      openContextMenu();
+
+      fixture.debugElement.query(By.directive(CdkMenuItem)).injector.get(CdkMenuItem).trigger();
+      fixture.detectChanges();
+
+      expect(getContextMenu()).not.toBeDefined();
+    });
+
+    it('should re-open the menu when right clicking twice in the context', () => {
```

Comment:
This test description leaves it ambiguous whether one or two menus are expected. The context string below does make it clearer, but maybe say 're-open the *same* menu' for extra clarity.